src/som/GraalSOMLanguage.java

```diff
@@ -0,0 +1,104 @@
+package som;
+
+import com.oracle.truffle.api.CallTarget;
+import com.oracle.truffle.api.Option;
+import com.oracle.truffle.api.Truffle;
+import com.oracle.truffle.api.TruffleLanguage;
+import com.oracle.truffle.api.debug.DebuggerTags;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.instrumentation.ProvidedTags;
+import com.oracle.truffle.api.instrumentation.StandardTags;
+import com.oracle.truffle.api.nodes.RootNode;
+
+import org.graalvm.options.OptionCategory;
+import org.graalvm.options.OptionDescriptors;
+import org.graalvm.options.OptionKey;
+import som.compiler.ProgramDefinitionError;
+import som.vm.Universe;
+import som.vmobjects.SAbstractObject;
+
+@TruffleLanguage.Registration(id = GraalSOMLanguage.ID,
+        name = "GraalSOM",
+        defaultMimeType = GraalSOMLanguage.MIME_TYPE,
+        characterMimeTypes = GraalSOMLanguage.MIME_TYPE,
+        contextPolicy = TruffleLanguage.ContextPolicy.SHARED,
+        fileTypeDetectors = GSFileDetector.class)
+@ProvidedTags({StandardTags.CallTag.class,
+        StandardTags.StatementTag.class,
+        StandardTags.RootTag.class,
+        StandardTags.RootBodyTag.class,
+        StandardTags.ExpressionTag.class,
+        DebuggerTags.AlwaysHalt.class})
+public final class GraalSOMLanguage extends TruffleLanguage<Universe> {
+
+    public static final String ID = "GS";
+    public static final String MIME_TYPE = "application/x-graal-som";
+    public static String[] args;
+
+    @Option(help = "Help Text.", category = OptionCategory.USER) public static final OptionKey<String> TestSelector = new OptionKey<>("");
+    @Option(help = "Help Text.", category = OptionCategory.USER) public static final OptionKey<String> TestClass = new OptionKey<>("");
+    @Option(help = "Help Text.", category = OptionCategory.USER) public static final OptionKey<String> TestClasspath = new OptionKey<>("");
+
+    public GraalSOMLanguage() {
+    }
+
+    @Override
+    protected Universe createContext(final Env env) {
+        return new Universe(env, this);
+    }
+
+    @Override
+    protected void initializeContext(final Universe context) {
+        Env currentEnv = context.getEnv();
```

Comment:
For simplicity, and to stay as close as possible to the other SOM implementations, I would use the same approach as SOMns. Simply letting the language take care of the various bits itself, and ignore the truffle framework.

Currently, I don't think we plan to look into language interop. So, I think it's perfectly fine to take the simple way out.

Reply:
Yes, I removed that part and the subsequent commented parts in the code about this matter. Let's keep to one `eval` per engine for now.